frontend/apps/app/features/projects/hooks/useProjectSearch.ts

```diff
@@ -0,0 +1,63 @@
+'use client'
+
+import { useState } from 'react'
+
+type Project = {
+  id: number
+  name: string
+  createdAt: string
+  organizationId: number | null
+}
+
+type SearchResult = {
+  projects: Project[] | null
+  loading: boolean
+  error: Error | null
+}
+
+export const useProjectSearch = (
+  organizationId?: number,
+  initialProjects?: Project[] | null,
+) => {
```

Comment:
I think I can get the types from ``supabase/database.types``

```suggestion
'use client'

import type { Tables } from '@liam-hq/db/supabase/database.types'
import { useState } from 'react'

type SearchResult = {
  projects: Tables<'Project'>[] | null
  loading: boolean
  error: Error | null
}

export const useProjectSearch = (
  organizationId?: number,
  initialProjects?: Tables<'Project'>[] | null,
) => {
```

In that case, though, you need to change `getProject.ts(line:6)` as follows.

```ts
  let query = supabase
    .from('Project')
    .select('id, name, createdAt, updatedAt, organizationId')
    .order('id', { ascending: false })
```